vunit/ui/results.py

```diff
@@ -24,5 +30,41 @@ def merge_coverage(self, file_name, args=None):
         :param file_name: The resulting coverage file name.
         :param args: The tool arguments for the merge command. Should be a list of strings.
         """
-
         self._simulator_if.merge_coverage(file_name=file_name, args=args)
+
+    def get_report(self):
+        """
+        Get a report (dictionary) of tests: status, time and output path
+        """
+
+        def get_relative_test_path(test_path):
+            """
+            If the test_path is relative to the default TEST_OUTPUT_PATH, return the subdir only
+            """
+            return re.sub(
+                "^%s" % normpath(join(self._output_path, TEST_OUTPUT_PATH)) + sep, "", normpath(test_path)
+            )
+
+        class Report(object):
+            """
+            Gives access to test results and paths after running tests.
+            """
+
+            def __init__(self, output_path):
+                self.output_path = output_path
+                self.tests = {}
+
+            def get_full_path(self, test_path, file_path=""):
```

Comment:
Since moving `get_full_path` requires to store `output_path` in each `ReportResult` object, I a took a different approach: `path` is saved raw in `ReportResult` (as @felixn preferred), and `report.get_relpath(test_path)` is provided.